crates/stages/src/stages/headers.rs

```diff
@@ -78,15 +78,16 @@ where
         mode: HeaderSyncMode,
         consensus: Arc<dyn Consensus>,
         tempdir: Arc<TempDir>,
+        etl_file_size: usize,
     ) -> Self {
         Self {
             provider: database,
             downloader,
             mode,
             consensus,
             sync_gap: None,
-            hash_collector: Collector::new(tempdir.clone(), 100 * (1024 * 1024)),
-            header_collector: Collector::new(tempdir, 100 * (1024 * 1024)),
+            hash_collector: Collector::new(tempdir.clone(), etl_file_size),
+            header_collector: Collector::new(tempdir, etl_file_size),
```

Comment:
@joshieDo what do you think if we do `etl_file_size / 2` here, so the ETL file size from config means how much max disk space to allocate for ETL files **per stage**?

Reply:
yeah, agree

Reply:
Done